### PR TITLE
Physical replication can insert, update, and delete. Use comdb2ar to …

### PR DESCRIPTION
**Pull request by kyleliangus**
This is an initial preview of upgrading physical replicants to use comdb2 transaction logs from the previous comdb2_oplog. 

There are changes in comdb2ar for easy setup of a local replicant. Currently, the supported features include insert, update, and deletion of records, and new file log changes. Schema changes and truncation is not yet supported.

### CODE REVIEW

**Comment by riverszhang89** [comment]
Please /runtests for Kyle.

File: bdb/phys_rep_lsn.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    DB_LOGC* logc;
+    DBT logrec;
+    DB_LSN last_log_lsn;
+    LOG_INFO log_info;
+    log_info.file = log_info.offset = log_info.size = 0;
+
+    rc = bdb_state->dbenv->log_cursor(bdb_state->dbenv, &logc, 0);
+    if (rc) {
+        logmsg(LOGMSG_ERROR, "%s: can't get log cursor rc %d\n", __func__, rc);
+        return log_info;
```
Comment (markhannum) [inline bdb/phys_rep_lsn.c]: no- log_info was declared on the stack- you can't return it.  Refactor this function to take a pointer to the returned lsn.  So, int get_last_lsn(bdb_state_type *bdb_state, LOG_INFO *loginfo)

File: db/phys_rep.c [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+
+    /* cannot call get_last_lsn if thedb is not ready */
+    while (!sc_ready())
+        nanosleep(&wait_spec, &remain_spec);
+
+    backend_thread_event(thedb, COMDB2_THR_EVENT_START_RDWR);
+
+    while(do_repl)
+    {
+        LOG_INFO info = get_last_lsn(thedb->bdb_env);
```
Comment (markhannum) [inline db/phys_rep.c]: Move the LOG_INFO declarations to above the while loop.

File: bdb/phys_rep_lsn.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    return log_info.offset + log_info.size + dbenv->get_log_header_size(dbenv);
+}
+
+int apply_log(DB_ENV* dbenv, int file, int offset, int64_t rectype, 
+        void* blob, int blob_len)
+{
+    return dbenv->apply_log(dbenv, file, offset, rectype, blob, blob_len);
+}
+
+int truncate_log(bdb_state_type* bdb_state, int file, int offset, int64_t rectype)
```
Comment (markhannum) [inline bdb/phys_rep_lsn.c]: You shouldn't need rectype- rep_verify_match doesn't use it.

**Comment by markhannum** [comment]
Closing- I've opened https://github.com/bloomberg/comdb2/pull/1103 for this

**Pull request closed without merging**